addons/mail/tests/discuss/test_discuss_channel.py

```diff
@@ -390,6 +390,24 @@ def test_channel_write_should_send_notification(self):
         ):
             channel.name = "test test"
 
+    def test_channel_mention_should_send_message_in_channel(self):
+        channel = self.env['discuss.channel'].create({"name": "Test Channel", "channel_type": "channel"})
+        partner = self.env["res.partner"].create({
+            "name": "Test Partner",
+            "email": "testpartner@odoo.com",
+        })
+        # Message Post
+        message = partner.message_post(body="Hello", channel_ids=[channel.id], message_type="comment", subtype_xmlid="mail.mt_comment")
+        # Check that the mentioned channel containt a message with id of channel
+        self.assertTrue(len(self.env['mail.message'].search([('res_id', '=', channel.id), ('model', '=', 'discuss.channel')])) == 1)
+        self.assertEqual(self.env['mail.message'].search([('id', '=', message.id), ('model', '=', 'res.partner')])['body'], '<p>Hello</p>')
```

Comment:
You can directly access the `body` property with `dot` notation instead of accessing this way.